### PR TITLE
New Pill module

### LINKED ISSUE #69: New Module: Pill
More details and spec to follow. @JeffNixonCreative can you please add?

This may end up being part of the button module, as to me it seems like just a rounded version of buttons and fake buttons.

screen shot 2018-02-26 at 10 31 10 am

Latest designs:
https://ebay.invisionapp.com/share/CWN9X1JF6MA#/screens/293357159

### PR DESCRIPTION
**Pull request by seangates**
## Description
- adds a `--pill` modifier

## Context
The new Pill module was added to the design system. It is added to both the DS4 and DS6 systems in order to allow for the component to be used by anyone who uses Skin.

## References
Closes #69 

## Screenshots
image

image

### CODE REVIEW

File: docs/src/less/ds4/index.less [hunk 1]
```diff
@@ -189,7 +189,10 @@ footer[role="contentinfo"] {
 .demo {
     background: #fff;
     border: 1px solid #eee;
-    word-break: break-all;
```
Comment (ianmcburnie) [inline docs/src/less/ds4/index.less]: Nice catch!

File: docs/src/less/ds4/index.less [hunk 2]
```diff
@@ -189,7 +189,10 @@ footer[role="contentinfo"] {
...
+
+    figure.highlight .s {
+        word-break: break-all;
```
Comment (ianmcburnie) [inline docs/src/less/ds4/index.less]: What is the `s` class? Not sure I've seen that been used.

Comment (seangates) [reply docs/src/less/ds4/index.less]: Again, this deals with the code highlighting features in the docs. I think `.s` means "strings" but I've found that this fixes the overflow problems in the examples (without this certain strings blow out the mobile web view of our pages).

Comment (ianmcburnie) [reply docs/src/less/ds4/index.less]: Oh, sorry. I was thinking about `span.highlight`, not `figure.highlight`.

Comment (seangates) [reply docs/src/less/ds4/index.less]: No worries.

**Comment by ianmcburnie** [comment]
Can we break the examples into subsections: "Default Pill" (with no state), "Multiline Pill", "Selected Pill", "Unselected Pill"? Then we can show the a11y guidance for each state (aria-pressed for button, clipped text for anchor).

**Pull request merged**